api/v1beta1/azureimage_validation.go

```diff
@@ -62,14 +65,45 @@ func validateSingleDetailsOnly(image *Image, fldPath *field.Path) field.ErrorLis
 
 	if image.SharedGallery != nil {
 		if imageDetailsFound {
-			allErrs = append(allErrs, field.Forbidden(fldPath.Child("SharedGallery"), "SharedGallery cannot be used as an image ID or Marketplace images has been specified"))
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("SharedGallery"), "SharedGallery cannot be used as an image ID, Marketplace or ComputeGallery images has been specified"))
+		} else {
+			imageDetailsFound = true
```

Comment:
Is it possible to add a [deprecation warning](https://kubernetes.io/blog/2020/09/03/warnings/#deprecation-warnings) when shared gallery is used?

Reply:
From what I see [kubebuilder only supports deprecating entire versions of objects](https://book.kubebuilder.io/reference/markers/crd.html) and generating custom warnings is currently not supported https://github.com/kubernetes-sigs/controller-runtime/issues/1896.

Reply:
I assume :+1: means there is nothing left to do here, at least for now. Will mark as resolved.